### PR TITLE
Fix AutogradMonkeyPatch typo, enable replacements

### PR DESCRIPTION
**Pull request by wconstab**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #89457
* #89802
* __->__ #89520

- seems like this codepath has not been exercised, and possibly is not needed
- running this through CI as a data point, but maybe we should delete this instead?

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_inductor/overrides.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ class AutogradMonkeypatch(TorchFunctionMode):
     def __torch_function__(self, func, types, args=(), kwargs=None):
         if not kwargs:
             kwargs = {}
-        if func is replacements:
+        if func in replacements:
```
Comment (wconstab) [inline torch/_inductor/overrides.py:31]: @jansel do you think this codepath should be needed?  or if things were working ok before should we just delete AutogradMonkeyPatch?

Comment (jansel) [reply torch/_inductor/overrides.py:31]: Ideally no, in practice maybe?

The dropout replacements was happening pretty inconsistently, so this was an (obviously broken) attempt to get them to work in a more robust way.  I think the bigger fix is to allow for pre-autograd replacements in AotAutograd.

**Review by jansel — APPROVED** [review]

**Comment by wconstab** [comment]
oops this got abandoned but someone else fixed it!
https://github.com/pytorch/pytorch/pull/90901
thanks!

**Pull request closed without merging**